x/oracle/keeper/upgrade.go

```diff
@@ -36,3 +38,32 @@ func (k Keeper) GetOracleUpgradeInfo(ctx sdk.Context) (*types.OracleUpgradeInfo,
 
 	return &upgradeInfo, nil
 }
+
+func (k Keeper) UpgradeOracle(ctx sdk.Context, msg *types.MsgUpgradeOracle) error {
+	oracleRegistration := msg.ToOracleRegistration()
+	compareFn := func(uniqueID string) error {
+		// check unique id
+		upgradeInfo, err := k.GetOracleUpgradeInfo(ctx)
+		if err != nil {
+			return err
+		}
+		if upgradeInfo.UniqueId != uniqueID {
+			return fmt.Errorf("The uniqueID to upgrade does not match. expected(%s), received(%s), ", upgradeInfo.UniqueId, msg.UniqueId)
+		}
```

Comment:
Block height should also be checked.
It should be greater than `upgradeInfo.Height`.
Because this `UpgradeOracle` may cause new vote, and if the vote is passed, it will update the oracle info with new uniqueID. Then, the oracle's vote might be invalid.
Please check if I'm right. @gyuguen

Reply:
I misunderstood. Please forget this comment. 

FYI, all the oracles should upgrade before reaching target height. If some oracles don't until target height, they have to re-register oracle with new version of oracle. [(ref comment)](https://github.com/medibloc/panacea-core/pull/454#discussion_r995379719)